Pythonwin/pywin/scintilla/control.py

```diff
@@ -472,7 +477,34 @@ def GetCurLineNumber(self):
     def GetTextLength(self):
         return self.SendScintilla(scintillacon.SCI_GETTEXTLENGTH)
 
-    def GetTextRange(self, start=0, end=-1, decode=True):
+    @overload
+    def GetTextRange(
+        self,
+        start: int,
+        end: int,
+        decode: Literal[False],
+    ) -> bytes: ...
+    @overload
+    def GetTextRange(
+        self,
+        start: int = 0,
+        end: int = -1,
+        *,
+        decode: Literal[False],
+    ) -> bytes: ...
+    @overload
+    def GetTextRange(
+        self,
+        start: int = 0,
+        end: int = -1,
+        decode: Literal[True] = True,
+    ) -> str: ...
```

Comment:
This overload makes it so callers statically know whether the return type will be `str` or `bytes` based on the `decode` param. (this fixes typing issues elsewhere in the code).